core/rwMRC.cpp

```diff
@@ -183,16 +185,22 @@ int ImageBase::readMRC(size_t select_img, bool isStack)
             _zDim = 1;
         }
 
-        if ( select_img > _nDim )
-            REPORT_ERROR(ERR_INDEX_OUTOFBOUNDS, formatString("readMRC: %s Image number %lu exceeds stack size %lu", this->filename.c_str(), select_img, _nDim));
+        if (batch_size == ALL_IMAGES) {
+            _nDimSet = _nDim - start_img + 1;
+        } else {
+        _nDimSet = std::min( start_img + batch_size - 1, _nDim ) - start_img + 1;
+        }
+
+
+        if ( start_img > _nDim )
+            REPORT_ERROR(ERR_INDEX_OUTOFBOUNDS, formatString("readMRC: %s Image number %lu exceeds stack size %lu", this->filename.c_str(), start_img, _nDim));
 
-        if (select_img > ALL_IMAGES)
-            _nDim = 1;
     }
     else // If the reading is not like a stack, then the select_img is not taken into account and must be selected the only image
     {
-        select_img = ALL_IMAGES;
+        start_img = 1;
```

Comment:
are you sure that it's 1?

Reply:
This part of if branch is visited only when `isStack==false`. The line  `_nDim=1;` is from devel, therefore the file contains only one image if isStack is false.